packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -626,6 +601,70 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	}
 
+	private createRevertible(
+		revision: RevisionTag,
+		kind: CommitKind,
+		checkout: TreeCheckout = this,
```

Comment:
I would avoid using default/optional parameters for checkout and onRevertibleDisposed here. Optionality is for providing convenience to the caller of the API. It can be especially important for public APIs, because that convenience scales according to the number of users (e.g. saving 10 seconds of coding time and 10 characters of source code per call = big impact if you have a million customers). But here, this API is private, it's only called twice, and in both places the parameters are explicitly provided anyway. We don't need the convenience, and it can actually be harmful (it allows callers to be careless and ignore the parameters). Having them be required paramaters forces callers of this method to think about which checkout they will pass, and what should happen when the revertible is disposed - that is a good thing for a private method that is only going to be touched by a couple of devs and which we want to be used thoughtfully and correctly.

Reply:
This is a great suggestion 😃. 

I am curious why [onRevertibleDisposed](https://github.com/microsoft/FluidFramework/blob/3c2775135055b4cc82f26796b83fc929e2bbee05/packages/dds/tree/src/shared-tree/treeCheckout.ts#L528) was optional in the first place. My impression is that it was set as optional to avoid unnecessary implementation of disposal functions in test suites (Based on an observation of `getRevertible()`'s usage in [treeCheckout.spec.ts](https://github.com/microsoft/FluidFramework/blob/3c2775135055b4cc82f26796b83fc929e2bbee05/packages/dds/tree/src/test/shared-tree/treeCheckout.spec.ts#L1018)). 

Since `createRevertible()` is responsible of creation of `RevertibleAlpha` object, I believe you are implying that I should also change the optional parameter in `getRevertible()`.  In this case do you think this PR should include changes to test-suites (although theses changes aren't directly relevant to Clonable Revertible)?

Reply:
No - not at all. All I'm saying is that you should change the private internal method.

Think of your audience and weigh the pros and cons.

Do you think your end-users, your customers of FF and SharedTree, will want to have to pass in a dummy dispose function, every time they call `getRevertible()`? Probably not. How many of them will be annoyed by that, because they don't care about listening to disposal? Probably most or all of them. Is making the parameter have a default going to lead them to write bugs? I don't think so - so there's no justification here for making them bothered.

Now think about the users of the private `createRevertible` function (that's you and me, and any other devs working on this project). Do you want any devs who are changing this code to completely understand what they are doing? Yes. Do you want them to consider all the implications of any change they make? Definitely. Is adding defaults going to help with that? No - it's going to allow us to forget things. Forcing us to provide the arguments forces us to think about what they should be. This is far more important than any convenience offered by eliminating a few characters from a method that is called only twice, ever.

That's the rationale. Depending on your audience, how many times the function is called, etc. etc. it might make sense to have defaults or not have defaults. But I think that in this instance, the right thing to do is pretty clear.

Reply:
This guidance would be great to add to our best practices documentation for parameter defaults.

Reply:
I added a section 👍